packages/core/src/batch/BatchRenderer.js

```diff
@@ -169,21 +169,37 @@ export default class BatchRenderer extends ObjectRenderer
 
     getIndexBuffer(size)
     {
-        const roundedSize = Math.ceil(size / 100.0) * 100;
+        // 12 indices is enough for 2 quads
+        const roundedP2 = nextPow2(Math.ceil(size / 12));
+        const roundedSizeIndex = log2(roundedP2);
+        const roundedSize = roundedP2 * 12;
 
-        let buffer = this.iBuffers[roundedSize];
+        while (this.iBuffers.length < roundedSizeIndex)
+        {
+            this.iBuffers.push(null);
+        }
```

Comment:
I think this is off-by-one, should be `<=` or it will not have the index `roundedSizeIndex`.

Instead can we just set the length?

```js
if (this.iBuffers.length <= roundedSizeIndex)
    this.iBuffers.length = roundedSizeIndex + 1;
```

Reply:
You're right but it actually doesn't matter, I read somewhere that setting a[a.length] is same as push. I'll fix it

Reply:
For example, using push:

![image](https://user-images.githubusercontent.com/944497/49887154-6bf50c00-fdf0-11e8-8c5f-63cdf2505aba.png)

And using `.length =`:

![image](https://user-images.githubusercontent.com/944497/49887176-7b745500-fdf0-11e8-99d5-26e471b6daae.png)

There is a difference in the engine, even if the JS user-experience is different only in that you get `null` rather than `undefined`. I'd prefer `.length=` because it is easier to read, smaller, and can be faster.

Reply:
i added `=` back then.

Ok, its `length=` now :)